Aspire Budgeting/Facilities/GoogleDriveManager.swift

```diff
@@ -9,35 +9,46 @@
 import Foundation
 import GoogleAPIClientForREST
 import GoogleSignIn
+import GTMSessionFetcher
 
 class GoogleDriveManager: ObservableObject {
-  private let driveService = GTLRDriveService()
+  static let queryFields: String = "kind,nextPageToken,files(mimeType,id,kind,name,webViewLink,thumbnailLink,trashed)"
   
+  private let driveService: GTLRService
   private var ticket: GTLRServiceTicket?
+  private let query: GTLRDriveQuery_FilesList
   
-  @Published var fileList = [File]()
-  @Published var error: Error?
+  @Published public private(set) var fileList = [File]()
+  @Published public private(set) var error: Error?
   
-  func getFileList(user: GIDGoogleUser) {
+  init(driveService: GTLRService = GTLRDriveService(),
+       query: GTLRDriveQuery_FilesList = GTLRDriveQuery_FilesList.query()) {
+    self.driveService = driveService
+    self.query = query
+  }
+  
+  func getFileList(authorizer: GTMFetcherAuthorizationProtocol) {
+    
+    let backupFileList = fileList
     fileList.removeAll()
     
-    let query = GTLRDriveQuery_FilesList.query()
-    driveService.authorizer = user.authentication.fetcherAuthorizer()
+    driveService.authorizer = authorizer
     driveService.shouldFetchNextPages = true
-    driveService.maxRetryInterval = 3
     
-    query.fields = "kind,nextPageToken,files(mimeType,id,kind,name,webViewLink,thumbnailLink,trashed)"
-    ticket = driveService.executeQuery(query, completionHandler: { [unowned self] (ticket, driveFileList, error) in
+    query.fields = GoogleDriveManager.queryFields
+    ticket = driveService.executeQuery(query, completionHandler: { [weak self] (ticket, driveFileList, error) in
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/mohitathwani/aspirebudgeting/blob/c2420c127e412148682bc0fe81f219ec96ee8752/Aspire Budgeting/Facilities/GoogleDriveManager.swift#L39"></span>Unused parameter “ticket” in a closure should be replaced with _.<br />
<code>unused_closure_parameter</code> <code>GoogleDriveManager.swift:39</code></td>
    </tr>
  </tbody>
</table>